packages/core/stories/toolkit-provider.doc.stories.mdx

```diff
@@ -96,6 +96,99 @@ It requires no configuration, but is included in your app when you use a Toolkit
 AriaAnnouncer is ever required, only the outermost instance of ToolkitProvider will instantiate the AriaAnnouncer. If you use
 nested ToolkitProviders, these will not inject further instances of AriaAnnouncer.
 
+## useDensity
+
+You can use `useDensity` hook to access the currently active density of your app in the provided scope.
+
+### API
+
+```js
+import { useDensity } from "@jpmorganchase/uitk-core";
+```
+
+### Example use
+
+Use density hook returns the string value of the currently set density in scope. You can optionally override the value bu passing your own value as the first parameter to the function.
+
+```tsx
+import { useDensity } from "@jpmorganchase/uitk-core";
+import cx from "classnames";
+
+function MyComponent() {
+  const [override, setDensityOverride] = useState();
+  const density = useDensity(override);
+  return (
+    <>
+      <div
+        className={cx("myComponent", {
+          "myComponent--touch": density === "touch",
+        })}
+      >
+        Hello World!
+      </div>
+      <button
+        onClick={() => {
+          setOverride("touch");
+        }}
+      >
+        Set Touch Density
+      </button>
+    </>
+  );
+}
+```
+
+## useTheme
+
+You can use `useTheme` hook to access the value of the currently active themes.
+
+```tsx
+import { useTheme } from "@jpmorganchase/uitk-core";
+
+function MyComponent() {
+  const [currentTheme] = useTheme();
+  return (
+    <div>{currentTheme === "uitk-light" ? "Light Theme" : "Dark Theme"}</div>
+  );
+}
+```
+
+### Example use
+
+Use density hook returns the string value of the currently set density in scope. You can optionally override the value bu passing your own value as the first parameter to the function.
+
+```tsx
+import { useDensity } from "@jpmorganchase/uitk-core";
+import cx from "classnames";
+
+function MyComponent() {
+  const [override, setDensityOverride] = useState();
+  const density = useDensity(override);
+  return (
+    <>
+      <div
+        className={cx("myComponent", {
+          "myComponent--touch": density === "touch",
+        })}
+      >
+        Hello World!
+      </div>
+      <button
+        onClick={() => {
+          setOverride("touch");
+        }}
+      >
+        Set Touch Density
+      </button>
+    </>
+  );
+}
+```
+
+## useBreakpoints
+
+You can use `useBreakpoints` hook to access the current value of `breakpoints` ToolkitProvider context. This is a low-level hook that is used internally in other hooks such as useResponsiveProp.
```

Comment:
There isn't any example of passing `breakpoints` right now, shall we add some? And reference components which are needed

Reply:
I couldn't imagine a practical example to demonstrate. Do you have any?

>And reference components which are needed

What do you mean by this?

Reply:
It is hard to make this hook seem useful without pulling in other hooks like useCurrentBreakpoint. But if you don't mind I could do something with that maybe 🤷

Reply:
Ok, that sounds good, but the layout component part. The issue there is that no component uses this hook directly. The layout components use higher level hooks like useResponsiveProp, or useCurrentBreakpoint.

Reply:
Maybe mention layout under generic 'breakpoint' heading, not specifically with this hook? Just need something, better than nothing

Reply:
Ok, So I added a dedicated breakpoints prop section where I mention the responsive props and layout components, and expanded the `useBreakpoints` section. Let me know if this version works.